utils/clock.py

```diff
@@ -0,0 +1,13 @@
+class Clock(object):
```

Comment:
How easy is this to extend to make it so different validators clocks move at different "rates?" Not sure if this question even makes sense, I just want to make sure we aren't forced to assume that validators clocks are synchronized.

Reply:
The Clock is actually pretty dumb as currently designed. The idea is to link it up to whatever environment it's in (single thread, independent thread, or even it's own process) and have the environment update the Clock. It's just the common interface for our objects to ask about time without having to care about the environment.